### DISCUSSION TITLE
Jupyter cell pending forever - Didn't get response for requestKernelInfo

### DISCUSSION THREAD

**Discussion by Pierre-LaurentC** [microsoft/vscode-jupyter#15040] [Questions and Answers] [answered]
I've recently experienced cell execution pending forever, with no modification from my side on the Jupyter extension or the Python virtual env used as kernel. I've done research about similar issues but the logs seem to differ, and the given fixes didn't work. Here is the trace of the Jupyter verbose output:

Visual Studio Code (1.83.0, ssh-remote, desktop)
Jupyter Extension Version: 2023.10.1100000000.
Python Extension Version: 2023.22.1.
Platform: linux (x64).
Workspace folder /data/plcristille, Home = /profils/plcristille
16:15:34.869 [debug] Get Custom Env Variables, Class name = Ul (started execution), Arg 1: undefined, Arg 2: "RunPythonCode"
16:15:34.870 [info] Start refreshing Kernel Picker (1705677334870)
16:15:34.874 [info] Using Pylance
16:15:35.587 [debug] No controller, hence notebook communications cannot be initialized for editor /data/<username>/Data/Sentinel_data_processing.ipynb
16:15:35.587 [info] Start refreshing Interpreter Kernel Picker
16:15:36.103 [debug] Get Custom Env Variables, Class name = Ul, completed in 1234ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunPythonCode"
16:15:36.103 [debug] Jupyter Paths /kernels: 
16:15:36.108 [debug] Kernel Spec Root Paths, /usr/share/jupyter/kernels, /usr/local/share/jupyter/kernels, ~/.local/share/jupyter/kernels
16:15:36.147 [debug] Search for KernelSpecs in Interpreter /data/<username>/virtual/dl/bin/python3.8
16:15:36.148 [debug] Finding Global Python KernelSpecs
16:15:36.150 [debug] Search for KernelSpecs in Interpreter /bin/python
16:15:36.150 [debug] Search for KernelSpecs in Interpreter /usr/bin/python
16:15:36.151 [debug] Search for KernelSpecs in Interpreter /usr/bin/python2
16:15:36.151 [debug] Search for KernelSpecs in Interpreter /bin/python2
16:15:36.152 [debug] Search for KernelSpecs in Interpreter /data/<username>/anaconda3/bin/python
16:15:36.152 [debug] Search for KernelSpecs in Interpreter ~/.conda/envs/copernicus/bin/python
16:15:36.153 [debug] Search for KernelSpecs in Interpreter /data/<username>/test/bin/python
16:15:36.154 [debug] Full interpreter list is length: 8, /data/<username>/virtual/dl/bin/python3.8:dl:Venv:/data/<username>/virtual/dl/bin/python3.8, /bin/python::Unknown:/bin/python, /usr/bin/python::Unknown:/usr/bin/python, /usr/bin/python2::Unknown:/usr/bin/python2, /bin/python2::Unknown:/bin/python2, /data/<username>/anaconda3/bin/python:base:Conda:/data/<username>/anaconda3/bin/python, ~/.conda/envs/copernicus/bin/python:copernicus:Conda:~/.conda/envs/copernicus/bin/python, /data/<username>/test/bin/python:test:Venv:/data/<username>/test/bin/python
16:15:36.155 [debug] Getting activated env variables, Class name = $a (started execution), Arg 1: "/data/<username>", Arg 2: "/data/<username>/virtual/dl/bin/python3.8", Arg 3: undefined
16:15:36.155 [debug] Get Custom Env Variables, Class name = Ul (started execution), Arg 1: "/data/<username>", Arg 2: "RunPythonCode"
16:15:36.156 [debug] Get Custom Env Variables, Class name = Ul, completed in 1ms, has a truthy return value, Arg 1: "/data/<username>", Arg 2: "RunPythonCode"
16:15:36.198 [debug] Kernels for interpreter /bin/python are .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
16:15:36.199 [debug] Kernels for interpreter ~/.conda/envs/copernicus/bin/python are .jvsc74a57bd06a9432f982e2f34aaa9740d43c1713ac3e49345f6fd08c7a29df11e53b4ceac7.~/.conda/envs/copernicus/python.~/.conda/envs/copernicus/python.-m#ipykernel_launcher
16:15:36.199 [debug] Kernels for interpreter /data/<username>/test/bin/python are .jvsc74a57bd06807e4e094ab0f3caa4520306f65b4dffd0eac54223e2d865a9315ed470f0517./data/<username>/test/python./data/<username>/test/python.-m#ipykernel_launcher
16:15:36.200 [debug] Kernels for interpreter /usr/bin/python are .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher
16:15:36.200 [debug] Kernels for interpreter /usr/bin/python2 are .jvsc74a57bd0f92b2924b84ff19c1c3dc485f7644d4486f64738191026bf8e6de303969141b5./usr/bin/python2./usr/bin/python2.-m#ipykernel_launcher
16:15:36.200 [debug] Kernels for interpreter /bin/python2 are .jvsc74a57bd0e55666fbbf217aa3df372b978577f47b6009e2f78e2ec76a584f49cd54a1e62c./bin/python2./bin/python2.-m#ipykernel_launcher
16:15:36.205 [debug] Loading kernelspec from /usr/share/jupyter/kernels/python3/kernel.json 
16:15:36.205 [debug] Loading kernelspec from /data/<username>/virtual/dl/share/jupyter/kernels/python3/kernel.json for /data/<username>/virtual/dl/bin/python3.8
16:15:36.206 [debug] Loading kernelspec from /data/<username>/anaconda3/share/jupyter/kernels/python3/kernel.json for /data/<username>/anaconda3/bin/python
16:15:36.223 [debug] Kernel Specs found in interpreter /data/<username>/virtual/dl/bin/python3.8 are [{"specFile":"/data/<username>/virtual/dl/share/jupyter/kernels/python3/kernel.json","interpreterPath":"/data/<username>/virtual/dl/bin/python3.8","isRegisteredByVSC":"registeredByNewVersionOfExt","name":"pythonjvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484","argv":["python","-m","ipykernel_launcher","-f","{connection_file}"],"language":"python","executable":"python","display_name":"Python 3 (ipykernel)","metadata":{"debugger":true,"vscode":{"originalSpecFile":"/data/<username>/virtual/dl/share/jupyter/kernels/python3/kernel.json","originalDisplayName":"Python 3 (ipykernel)"}}}]
16:15:36.223 [debug] Hiding default kernel spec 'Python 3 (ipykernel)', 'pythonjvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484', /python for interpreter /data/<username>/virtual/dl/bin/python3.8 and spec /data/<username>/virtual/dl/share/jupyter/kernels/python3/kernel.json
16:15:36.223 [debug] Kernels for interpreter /data/<username>/virtual/dl/bin/python3.8 are .jvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484./data/<username>/virtual/dl/bin/python3.8./data/<username>/virtual/dl/bin/python3.8.-m#ipykernel_launcher
16:15:36.223 [debug] Kernel Specs found in interpreter /data/<username>/anaconda3/bin/python are [{"specFile":"/data/<username>/anaconda3/share/jupyter/kernels/python3/kernel.json","interpreterPath":"/data/<username>/anaconda3/bin/python","isRegisteredByVSC":"registeredByNewVersionOfExt","name":"pythonjvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada","argv":["/data/<username>/anaconda3/bin/python","-m","ipykernel_launcher","-f","{connection_file}"],"language":"python","executable":"/data/<username>/anaconda3/bin/python","display_name":"Python 3 (ipykernel)","metadata":{"debugger":false,"vscode":{"originalSpecFile":"/data/<username>/anaconda3/share/jupyter/kernels/python3/kernel.json","originalDisplayName":"Python 3 (ipykernel)"}}}]
16:15:36.223 [debug] Hiding default kernel spec 'Python 3 (ipykernel)', 'pythonjvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada', /data/<username>/anaconda3/bin/python for interpreter /data/<username>/anaconda3/bin/python and spec /data/<username>/anaconda3/share/jupyter/kernels/python3/kernel.json
16:15:36.224 [debug] Kernels for interpreter /data/<username>/anaconda3/bin/python are .jvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada./data/<username>/anaconda3/python./data/<username>/anaconda3/python.-m#ipykernel_launcher
16:15:36.295 [debug] KernelProvider switched kernel to id = .jvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484./data/<username>/virtual/dl/bin/python3.8./data/<username>/virtual/dl/bin/python3.8.-m#ipykernel_launcher
16:15:36.296 [debug] start the kernel, options.disableUI=true for /data/<username>/Data/Sentinel_data_processing.ipynb
16:15:36.297 [debug] KernelProvider switched kernel to id = .jvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada./data/<username>/anaconda3/python./data/<username>/anaconda3/python.-m#ipykernel_launcher
16:15:36.297 [debug] start the kernel, options.disableUI=true for /data/<username>/Data/S2_SRF_Reader.ipynb
16:15:36.304 [debug] Initialize notebook communications for editor /data/<username>/Data/Sentinel_data_processing.ipynb
16:15:36.304 [debug] Resolving notebook UI Comms (resolve) for /data/<username>/Data/Sentinel_data_processing.ipynb
16:15:36.304 [debug] initialize CommonMessageCoordinator
16:15:36.305 [debug] Registering commtarget jupyter.widget
16:15:36.305 [debug] IPyWidgetMessageDispatcher.initialize
16:15:36.305 [debug] Attempting to determine version of IPyWidgets
16:15:36.306 [debug] Controller selection change completed
16:15:36.306 [debug] Controller selection change completed
16:15:36.306 [debug] Waiting for IPyWidgets version
16:15:36.306 [debug] Waiting for IPyWidgets version promise
16:15:36.381 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484./data/<username>/virtual/dl/bin/python3.8./data/<username>/virtual/dl/bin/python3.8.-m#ipykernel_launcher  (Python Path: /data/<username>/virtual/dl/bin/python3.8, Venv, dl, 3.8.10) for '/data/<username>/Data/Sentinel_data_processing.ipynb' (disableUI=true)
16:15:36.382 [debug] Creating raw notebook for resource '/data/<username>/Data/Sentinel_data_processing.ipynb'
16:15:36.383 [debug] Getting activated env variables, Class name = $a (started execution), Arg 1: "/data/<username>", Arg 2: "/data/<username>/anaconda3/bin/python", Arg 3: undefined
16:15:36.383 [debug] Get Custom Env Variables, Class name = Ul (started execution), Arg 1: "/data/<username>", Arg 2: "RunPythonCode"
16:15:36.383 [debug] Get Custom Env Variables, Class name = Ul, completed in 0ms, has a truthy return value, Arg 1: "/data/<username>", Arg 2: "RunPythonCode"
16:15:36.383 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada./data/<username>/anaconda3/python./data/<username>/anaconda3/python.-m#ipykernel_launcher  (Python Path: /data/<username>/anaconda3/bin/python, Conda, base, 3.9.18) for '/data/<username>/Data/S2_SRF_Reader.ipynb' (disableUI=true)
16:15:36.383 [debug] Creating raw notebook for resource '/data/<username>/Data/S2_SRF_Reader.ipynb'
16:15:36.393 [debug] Kernel Launcher. launch, Class name = Fv (started execution), Arg 1: ".jvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484./data/<username>/virtual/dl/bin/python3.8./data/<username>/virtual/dl/bin/python3.8.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 4: "/data/<username>/Data", Arg 5: ""
16:15:36.402 [debug] Getting activated env variables, Class name = $a (started execution), Arg 1: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 2: "/data/<username>/virtual/dl/bin/python3.8", Arg 3: undefined
16:15:36.402 [debug] Get Custom Env Variables, Class name = Ul (started execution), Arg 1: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 2: "RunPythonCode"
16:15:36.402 [debug] Get Custom Env Variables, Class name = Ul, completed in 0ms, has a truthy return value, Arg 1: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 2: "RunPythonCode"
16:15:36.405 [debug] Kernel Launcher. launch, Class name = Fv (started execution), Arg 1: ".jvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada./data/<username>/anaconda3/python./data/<username>/anaconda3/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 4: "/data/<username>/Data", Arg 5: ""
16:15:36.406 [debug] Getting activated env variables, Class name = $a (started execution), Arg 1: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 2: "/data/<username>/anaconda3/bin/python", Arg 3: undefined
16:15:36.406 [debug] Get Custom Env Variables, Class name = Ul (started execution), Arg 1: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 2: "RunPythonCode"
16:15:36.407 [debug] Get Custom Env Variables, Class name = Ul, completed in 1ms, has a truthy return value, Arg 1: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 2: "RunPythonCode"
16:15:36.466 [debug] Launching kernel in kernelProcess.ts, Class name = Lv (started execution), Arg 1: "/data/<username>/Data", Arg 2: ""
16:15:36.502 [debug] Launching kernel in kernelProcess.ts, Class name = Lv (started execution), Arg 1: "/data/<username>/Data", Arg 2: ""
16:15:36.783 [debug] Got env vars with python /data/<username>/virtual/dl/bin/python3.8, with env var count 35 in 628ms. 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.783 [debug] Prepend PATH with python bin for /data/<username>/virtual/dl/bin/python3.8, 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.784 [debug] Activated Env Variables for /data/<username>/virtual/dl/bin/python3.8, 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.784 [debug] Getting activated env variables, Class name = $a, completed in 629ms, has a truthy return value, Arg 1: "/data/<username>", Arg 2: "/data/<username>/virtual/dl/bin/python3.8", Arg 3: undefined
16:15:36.807 [info] Process Execution: /data/<username>/virtual/dl/bin/python3.8 -m pip list
16:15:36.809 [debug] Got env vars with python /data/<username>/virtual/dl/bin/python3.8, with env var count 35 in 407ms. 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.809 [debug] Prepend PATH with python bin for /data/<username>/virtual/dl/bin/python3.8, 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.809 [debug] Activated Env Variables for /data/<username>/virtual/dl/bin/python3.8, 
    PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:36.810 [debug] Getting activated env variables, Class name = $a, completed in 408ms, has a truthy return value, Arg 1: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 2: "/data/<username>/virtual/dl/bin/python3.8", Arg 3: undefined
16:15:36.831 [info] Process Execution: /data/<username>/virtual/dl/bin/python3.8 -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
16:15:36.831 [debug] Kernel Env Variables for ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.10.1100000000-linux-x64/temp/jupyter/kernels/pythonjvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484/kernel.json, PATH value is /data/<username>/virtual/dl/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin
16:15:36.844 [info] Process Execution: /data/<username>/virtual/dl/bin/python3.8 -m ipykernel_launcher --f=~/.local/share/jupyter/runtime/kernel-v2-715790HRDve2gU4wZI.json
    > cwd: /data/<username>/Data
16:15:36.844 [debug] Launching kernel in kernelProcess.ts, Class name = Lv, completed in 378ms, has a truthy return value, Arg 1: "/data/<username>/Data", Arg 2: ""
16:15:36.858 [debug] Conda file returned by Python Extension is conda
16:15:36.990 [info] ipykernel version & path 6.29.0, /data/<username>/virtual/dl/lib/python3.8/site-packages/ipykernel/__init__.py for /data/<username>/virtual/dl/bin/python3.8
16:15:37.183 [debug] Python API env change detected, update => '/data/<username>/virtual/dl/bin/python3.8'
16:15:37.183 [debug] Python API env change detected, update => '/data/<username>/anaconda3/bin/python'
16:15:37.184 [debug] Python API env change detected, update => '/data/<username>/test/bin/python'
16:15:37.204 [debug] Full interpreter list is length: 8, /data/<username>/virtual/dl/bin/python3.8:dl:Venv:/data/<username>/virtual/dl/bin/python3.8, /bin/python::Unknown:/bin/python, /usr/bin/python::Unknown:/usr/bin/python, /usr/bin/python2::Unknown:/usr/bin/python2, /bin/python2::Unknown:/bin/python2, /data/<username>/anaconda3/bin/python:base:Conda:/data/<username>/anaconda3/bin/python, /data/<username>/test/bin/python:test:Venv:/data/<username>/test/bin/python, ~/.conda/envs/copernicus/bin/python:copernicus:Conda:~/.conda/envs/copernicus/bin/python
16:15:37.204 [debug] Refreshed Environments
16:15:37.208 [debug] KernelProcess output: NOTE: When using the `ipython kernel` entry point, Ctrl-C will not work.

To exit, you will have to explicitly quit this process, by either sending
"quit" from a client, or using Ctrl-\ in UNIX-like environments.

To read more about this, see https://github.com/ipython/ipython/issues/2049

To connect another client to this kernel, use:
    --existing kernel-v2-715790HRDve2gU4wZI.json

16:15:37.209 [debug] Kernel Output: To connect another client to this kernel, use: --existing kernel-v2-715790HRDve2gU4wZI.json

16:15:37.274 [debug] Kernel Launcher. launch, Class name = Fv, completed in 881ms, has a truthy return value, Arg 1: ".jvsc74a57bd0f298dbb6a34056a7ea8eca852791b06fd80298f8dd2695303f14f1ff43226484./data/<username>/virtual/dl/bin/python3.8./data/<username>/virtual/dl/bin/python3.8.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "/data/<username>/Data/Sentinel_data_processing.ipynb", Arg 4: "/data/<username>/Data", Arg 5: ""
16:15:37.279 [debug] Waiting for Raw Session to be ready in postStartRawSession
16:15:37.279 [debug] Waiting for Raw session to be ready, currently connected
16:15:37.279 [debug] Raw session connected
16:15:37.279 [debug] Waiting for Raw session to be ready for 30s
16:15:37.280 [debug] Waited for Raw session to be ready & got connected
16:15:37.280 [debug] Successfully waited for Raw Session to be ready in postStartRawSession
16:15:37.280 [debug] Kernel status before requesting kernel info and after ready is unknown
16:15:37.280 [debug] Sending request for kernelInfo
16:15:37.476 [debug] Got env vars with python /data/<username>/anaconda3/bin/python, with env var count 41 in 1070ms. 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.476 [debug] Prepend PATH with python bin for /data/<username>/anaconda3/bin/python, 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.476 [debug] Activated Env Variables for /data/<username>/anaconda3/bin/python, 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.476 [debug] Getting activated env variables, Class name = $a, completed in 1070ms, has a truthy return value, Arg 1: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 2: "/data/<username>/anaconda3/bin/python", Arg 3: undefined
16:15:37.492 [info] Process Execution: /data/<username>/anaconda3/bin/python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
16:15:37.493 [debug] Kernel Env Variables for ~/.vscode-server/extensions/ms-toolsai.jupyter-2023.10.1100000000-linux-x64/temp/jupyter/kernels/pythonjvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada/kernel.json, PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin
16:15:37.509 [info] Process Execution: /data/<username>/anaconda3/bin/python -m ipykernel_launcher --f=~/.local/share/jupyter/runtime/kernel-v2-715790Gi2rdUm6zadR.json
    > cwd: /data/<username>/Data
16:15:37.509 [debug] Launching kernel in kernelProcess.ts, Class name = Lv, completed in 1007ms, has a truthy return value, Arg 1: "/data/<username>/Data", Arg 2: ""
16:15:37.513 [debug] Got env vars with python /data/<username>/anaconda3/bin/python, with env var count 41 in 1130ms. 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.513 [debug] Prepend PATH with python bin for /data/<username>/anaconda3/bin/python, 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.513 [debug] Activated Env Variables for /data/<username>/anaconda3/bin/python, 
    PATH value is /data/<username>/anaconda3/bin:~/.vscode-server/bin/e7e037083ff4455cf320e344325dacb480062c3c/bin/remote-cli:~/.local/bin:/data/<username>/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/snap/bin and 
    Path value is undefined
16:15:37.513 [debug] Getting activated env variables, Class name = $a, completed in 1130ms, has a truthy return value, Arg 1: "/data/<username>", Arg 2: "/data/<username>/anaconda3/bin/python", Arg 3: undefined
16:15:37.537 [info] Process Execution: /data/<username>/anaconda3/bin/python -m pip list
16:15:37.628 [info] ipykernel version & path 6.25.0, /data/<username>/anaconda3/lib/python3.9/site-packages/ipykernel/__init__.py for /data/<username>/anaconda3/bin/python
16:15:37.927 [debug] KernelProcess output: NOTE: When using the `ipython kernel` entry point, Ctrl-C will not work.

To exit, you will have to explicitly quit this process, by either sending
"quit" from a client, or using Ctrl-\ in UNIX-like environments.

To read more about this, see https://github.com/ipython/ipython/issues/2049

To connect another client to this kernel, use:
    --existing kernel-v2-715790Gi2rdUm6zadR.json

16:15:37.927 [debug] Kernel Output: To connect another client to this kernel, use: --existing kernel-v2-715790Gi2rdUm6zadR.json

16:15:37.946 [debug] Kernel Launcher. launch, Class name = Fv, completed in 1541ms, has a truthy return value, Arg 1: ".jvsc74a57bd0a8afb1b6a211558bc172f642a4e07327d53ea7dc9131a9786570087a19d11ada./data/<username>/anaconda3/python./data/<username>/anaconda3/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "/data/<username>/Data/S2_SRF_Reader.ipynb", Arg 4: "/data/<username>/Data", Arg 5: ""
16:15:37.947 [debug] Waiting for Raw Session to be ready in postStartRawSession
16:15:37.947 [debug] Waiting for Raw session to be ready, currently connected
16:15:37.947 [debug] Raw session connected
16:15:37.947 [debug] Waiting for Raw session to be ready for 30s
16:15:37.947 [debug] Waited for Raw session to be ready & got connected
16:15:37.947 [debug] Successfully waited for Raw Session to be ready in postStartRawSession
16:15:37.947 [debug] Kernel status before requesting kernel info and after ready is unknown
16:15:37.947 [debug] Sending request for kernelInfo
16:15:38.205 [info] End refreshing Kernel Picker (1705677334870)
16:15:38.780 [debug] Did not get a response for requestKernelInfo
16:15:38.780 [debug] Sending request for kernelInfo
16:15:39.448 [debug] Did not get a response for requestKernelInfo
16:15:39.448 [debug] Sending request for kernelInfo
16:15:40.279 [debug] Did not get a response for requestKernelInfo
16:15:40.279 [warn] Didn't get response for requestKernelInfo after 2999ms.
16:15:40.281 [debug] Begin sendTelemetryForPythonKernelExecutable
16:15:40.282 [debug] Executing silently Code (unknown) = import sys as _VSCODE_sys\nprint(_VSCODE_sys.executable); del _VSCODE_sys
16:15:40.948 [debug] Did not get a response for requestKernelInfo
16:15:40.948 [warn] Didn't get response for requestKernelInfo after 3001ms.
16:15:40.949 [debug] Begin sendTelemetryForPythonKernelExecutable
16:15:40.949 [debug] Executing silently Code (unknown) = import sys as _VSCODE_sys\nprint(_VSCODE_sys.executable); del _VSCODE_sys
16:15:50.181 [info] Handle Execution of Cells 0 for /data/<username>/Data/Sentinel_data_processing.ipynb
16:15:50.182 [debug] start the kernel, options.disableUI=false for /data/<username>/Data/Sentinel_data_processing.ipynb

**Comment by DonJayamanne** [comment, marked as answer]
Duplicate of https://github.com/microsoft/vscode-jupyter/issues/15051
